python-sqlite-sqlalchemy/project/examples/example_3/app/tracks/routes.py

```diff
@@ -0,0 +1,113 @@
+from flask import Blueprint
+from flask import render_template
+from flask_wtf import FlaskForm
+from wtforms import StringField
+from wtforms import IntegerField
+from wtforms import FloatField
+from wtforms import SelectField
+from wtforms.validators import InputRequired
+from wtforms.validators import ValidationError
+from app import db
+from app.models import Artist
+from app.models import Album
+from app.models import Track
+from app.models import MediaType
+from app.models import Genre
+
+
+# Setup the Blueprint
+tracks_bp = Blueprint(
+    "tracks_bp", __name__, template_folder="templates", static_folder="static"
+)
+
+
+def does_artist_exist(form, field):
+    artist = (
+        db.session.query(Artist)
+        .filter(Artist.name == field.data)
+        .one_or_none()
+    )
+
+    if artist is not None:
+        raise ValidationError("Artist already exists", field.data)
+
+
+def does_album_exist(form, field):
+    album = (
+        db.session.query(Album).filter(Album.title == field.data).one_or_none()
+    )
+
+    if album is not None:
+        raise ValidationError("Album already exists", field.data)
+
+
+def does_track_exist(form, field):
+    track = (
+        db.session.query(Track).filter(Track.name == field.data).one_or_none()
+    )
+
+    if track is not None:
+        raise ValidationError("Track already exists", field.data)
```

Comment:
I think this logic ends up being a little too simple. Since it only compares track titles, it will refuse to add any song named for instance "Crazy" for any artist on any album, just because Aerosmith is registered with a song named "Crazy" already.

Would it be possible to change this to look for actual duplicates instead? Where at least artist, album, and track name are reused?

Reply:
Similarly for the logic adding albums

Reply:
Done